File: 93.js
function findTwoUnique(nums) {
    let xor = 0;

    for (let n of nums) xor ^= n;

    // Rightmost set bit
    let rightBit = xor & -xor;

    let a = 0, b = 0;

    for (let n of nums) {
        if (n & rightBit) a ^= n;
        else b ^= n;
    }

    return [a, b];
}

// Test
console.log(findTwoUnique([4, 1, 2, 1, 2, 5]));

File: 93.py
def find_two_unique(nums):
    xor = 0
    for n in nums:
        xor ^= n

    right_bit = xor & -xor

    a, b = 0, 0
    for n in nums:
        if n & right_bit:
            a ^= n
        else:
            b ^= n

    return a, b

print(find_two_unique([4,1,2,1,2,5]))
